lib/Dialect/FIRRTL/Transforms/LowerXMR.cpp

```diff
@@ -262,6 +255,37 @@ class LowerXMRPass : public LowerXMRBase<LowerXMRPass> {
         if (transferFunc(op).failed())
           return signalPassFailure();
 
+      // Since we walk operations pre-order and not along dataflow edges,
+      // ref.sub may not be resolvable when we encounter them (they're not just
+      // unification). This can happen when refs go through an output port or
+      // input instance result and back into the design. Handle these by walking
+      // them, resolving what we can, until all are handled or nothing can be
+      // resolved.
+      while (!indexingOps.empty()) {
+        // Grab the set of unresolved ref.sub's.
+        decltype(indexingOps) worklist;
+        worklist.swap(indexingOps);
+
+        for (auto op : worklist) {
+          auto inputEntry = getRemoteRefSend(op.getInput());
```

Comment:
wouldn't `getRemoteRefSend` error out, if it can't resolve ?

Reply:
Whoops, yes it will.  Will fix, thanks!

Reply:
Good call, thank you! Updated!